pkg/vsphere/simulator/virtual_machine.go

```diff
@@ -299,6 +309,10 @@ func (vm *VirtualMachine) configureDevice(devices object.VirtualDeviceList, devi
 		devices.AssignController(device, controller)
 	}
 
+	if d.Key == -1 {
```

Comment:
Yes, I am planning to fix this within object.VirtualDeviceList.  In some cases it uses NewKey(), which returns a negative number unique to the list, in others it just uses -1.  So we're just making the Key unique on the vcsim side in that particular case.  vcsim also leaves the Key as a negative number currently, would also like to take care of that within VirtualDeviceList.